test/framework/integration_framework/fake_peer/fake_peer.cpp

```diff
@@ -0,0 +1,198 @@
+                  std::string>::ValueType &result) {
+            peer = std::move(result.value);
+          },
+          [&address](const iroha::expected::Result<
+              std::unique_ptr<shared_model::interface::Peer>,
+              std::string>::ErrorType &error) {
+            BOOST_THROW_EXCEPTION(
+                std::runtime_error("Failed to create peer object for peer "
+                                   + address + ". " + error.error));
+          });
+  return peer;
+}
+
+namespace integration_framework {
+
+  FakePeer::FakePeer(
+      const std::string &listen_ip,
+      size_t internal_port,
+      const boost::optional<Keypair> &key,
+      const std::shared_ptr<shared_model::interface::Peer> &real_peer,
+      const std::shared_ptr<shared_model::interface::CommonObjectsFactory>
+          &common_objects_factory,
+      std::shared_ptr<TransportFactoryType> transaction_factory,
+      std::shared_ptr<shared_model::interface::TransactionBatchParser>
+          batch_parser,
+      std::shared_ptr<shared_model::interface::TransactionBatchFactory>
+          transaction_batch_factory,
+      bool agree_all_proposals)
+      : common_objects_factory_(common_objects_factory),
+        listen_ip_(listen_ip),
+        internal_port_(internal_port),
+        keypair_(std::make_unique<Keypair>(
+            key.value_or(DefaultCryptoAlgorithmType::generateKeypair()))),
+        this_peer_(createPeer(
+            common_objects_factory, getAddress(), keypair_->publicKey())),
+        real_peer_(real_peer),
+        async_call_(std::make_shared<AsyncCall>()),
+        mst_transport_(std::make_shared<MstTransport>(async_call_,
+                                                      transaction_factory,
+                                                      batch_parser,
+                                                      transaction_batch_factory,
+                                                      keypair_->publicKey())),
+        yac_transport_(std::make_shared<YacTransport>(async_call_)),
+        yac_network_notifier_(std::make_shared<YacNetworkNotifier>()),
+        yac_crypto_(std::make_shared<iroha::consensus::yac::CryptoProviderImpl>(
+            *keypair_, common_objects_factory)) {
+    yac_network_notifier_->subscribe(yac_transport_);
+    log_ = logger::log(
+        "IntegrationTestFramework "
+        "(fake peer at "
+        + getAddress() + ")");
+    if (agree_all_proposals) {
+      enableAgreeAllProposals();
+    }
+  }
+
+  void FakePeer::run() {
+    // start instance
+    log_->info("starting listening server");
+    internal_server_ = std::make_unique<ServerRunner>(getAddress());
+    internal_server_->append(yac_transport_)
+        .append(mst_transport_)
+        .run()
+        .match(
+            [this](const iroha::expected::Result<int, std::string>::ValueType
+                       &val) {
+              log_->debug("started server on port {}", val.value);
```

Comment:
It seems to be printed not that often, so it might be better write at info level